### PR TITLE
Support editing lists, args, and kwargs

### PR DESCRIPTION
**Pull request by JosephMarinier**
## Description:

Screen Shot 2023-03-23 at 10 54 53 AM

## Checklist:

You should check all boxes before the PR is ready. If a box does not apply, check it to acknowledge it.

* [x] **ISSUE NUMBER.** You linked the issue number (Ex: Resolve #XXX).
* [x] **PRE-COMMIT.** You ran pre-commit on all commits, or else, you
  ran `pre-commit run --all-files` at the end.
* [x] **USER CHANGES.** The changes are added to CHANGELOG.md and the documentation, if they impact
  our users.
* [x] **DEV CHANGES.**
    * Update the documentation if this PR changes how to develop/launch on the app.
    * Update the `README` files and our wiki for any big design decisions, if relevant.
    * Add unit tests, docstrings, typing and comments for complex sections.

### CODE REVIEW

File: webapp/src/pages/Settings.tsx [hunk 1]
```diff
@@ -401,6 +478,26 @@ const Settings: React.FC<Props> = ({ open, onClose }) => {
     );
   }
 
+  // TODO Reuse that kind of handler to reduce duplication.
+  const updateModel = (
+    pipelineIndex: number,
+    update: Partial<PipelineDefinition["model"]>
+  ) =>
+    setPartialConfig({
```
Comment (christyler3030) [inline webapp/src/pages/Settings.tsx:486]: Might be safer to use 
`setPartialConfig(partialConfig => ({...partialConfig, etc...}))`
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

Comment (JosephMarinier) [reply webapp/src/pages/Settings.tsx:486]: Oh cool! I didn't know that existed. 👍 
It is like that in more than 20 other places, so I'll make another refactor PR to cover everything.

Comment (christyler3030) [reply webapp/src/pages/Settings.tsx:486]: If you decide to do the changes in this file in this PR there are a few more instances in this file as well.

**Review by christyler3030 — APPROVED** [review]
Approved with comments

File: webapp/src/pages/Settings.tsx (hunk 1 above)
Comment (JosephMarinier) [reply webapp/src/pages/Settings.tsx:486]: Yeah, yeah, I am just talking about this file: 21 instances of `setPartialConfig({...partialConfig`.

**Pull request merged**